modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonModulePlugin.java

```diff
@@ -165,7 +165,7 @@ private Map<String, Processor.Factory> filterForAllowlistSetting(Settings settin
         // Assert that no unknown processors are defined in the allowlist
         final Set<String> unknownAllowlistProcessors = allowlist.stream()
             .filter(p -> map.containsKey(p) == false)
-            .collect(Collectors.toSet());
+            .collect(Collectors.toUnmodifiableSet());
```

Comment:
@owaiskazi19 See what I did here? I said I'd address it in a follow up PR so now you're on the hook to review this one :)